packages/framework/src/Component/Elasticsearch/ElasticsearchStructureManager.php

```diff
@@ -102,4 +242,104 @@ protected function getDefinition(int $domainId, string $index): array
 
         return $definition;
     }
+
+    /**
+     * @param array $array
+     * @return bool
+     */
+    protected function recursiveArraySort(array &$array)
+    {
+        foreach ($array as &$value) {
+            if (is_array($value)) {
+                $this->recursiveArraySort($value);
+            }
+        }
+        return ksort($array);
+    }
+
+    /**
+     * @param array $currentIndex
+     * @return array
+     */
+    protected function prepareCurrentIndexDefinition(array $currentIndex): array
+    {
+        $currentIndexDefinition = reset($currentIndex);
+        $currentIndexDefinition['settings'] = $currentIndexDefinition['settings']['index'];
+        $currentIndexDefinition['settings']['index'] = [
+            'number_of_replicas' => (int)$currentIndexDefinition['settings']['number_of_replicas'],
+            'number_of_shards' => (int)$currentIndexDefinition['settings']['number_of_shards'],
+        ];
+        $currentIndexDefinition['settings']['analysis']['filter']['edge_ngram']['max_gram'] =
+            (int)$currentIndexDefinition['settings']['analysis']['filter']['edge_ngram']['max_gram'];
+        $currentIndexDefinition['settings']['analysis']['filter']['edge_ngram']['min_gram'] =
+            (int)$currentIndexDefinition['settings']['analysis']['filter']['edge_ngram']['min_gram'];
+        unset(
+            $currentIndexDefinition['aliases'],
+            $currentIndexDefinition['settings']['provided_name'],
+            $currentIndexDefinition['settings']['creation_date'],
+            $currentIndexDefinition['settings']['uuid'],
+            $currentIndexDefinition['settings']['version'],
+            $currentIndexDefinition['settings']['number_of_replicas'],
+            $currentIndexDefinition['settings']['number_of_shards']
+        );
+        return $currentIndexDefinition;
+    }
+
+    /**
+     * @param int $domainId
+     * @param string $index
+     */
+    public function createIndexIfNecessary(int $domainId, string $index)
+    {
+        if ($this->isNecessaryToUpdateStructure($domainId, $index)) {
+            $this->createIndex($domainId, $index);
+        }
+    }
+
+    /**
+     * @param int $domainId
+     * @param string $index
+     */
+    public function createIndexAndReindexIfNecessary(int $domainId, string $index)
+    {
+        if ($this->isNecessaryToUpdateStructure($domainId, $index)) {
+            $this->createIndex($domainId, $index, false);
+            $this->reindex($domainId, $index);
+            $this->deleteNotUsedIndexesOfCurrentAlias($domainId, $index);
+            $this->createAliasForCurrentIndex($domainId, $index);
+        }
+    }
+
+    /**
+     * @param int $domainId
+     * @param string $index
+     * @return bool
+     */
+    protected function isNecessaryToUpdateStructure(int $domainId, string $index)
```

Comment:
After all I have a strong feeling that the `ElasticStructureManager` is doing too much now and I don't understand what is it's responsibility. Maybe that's because it contains functionality that should be separated somewhere else